scylla/src/transport/session.rs

```diff
@@ -606,6 +613,10 @@ impl Session {
             }
         }
 
+        if self.refresh_metadata_on_schema_agreement && response.as_schema_change().is_some() {
```

Comment:
The name of the flag is `refresh_metadata_on_schema_agreement`. However, I'm not sure its name aligns with what it really does:

- If automatic wait for schema agreement is disabled and `refresh_metadata_on_schema_agreement` is true, the driver will still refresh metadata, even though it didn't wait for schema agreement.
- If the user explicitly waits for schema agreement (`session.await_schema_agreement()`), metadata won't be refreshed after it finishes.

Perhaps we should move the wait into `await_schema_agreement` in order to align its behavior with the name. @cvybhu what do you think?

Reply:
Maybe it would be a better idea to name it `refresh_metadata_on_auto_schema_agreement` and make it to refresh metadata only if the automatic wait for schema agreement is enabled. I think it would be better that the user explicitly calls `refresh_metadata` after manually waiting for schema agreement.

Reply:
Yes, this is also a valid option.

Reply:
The problem from the first bullet point is still there.

Reply:
Fixed it, tnx.